Lib/typing.py

```diff
@@ -2925,6 +2927,42 @@ def __round__(self, ndigits: int = 0) -> T:
         pass
 
 
+@runtime_checkable
+class Reader[T](Iterable[T], Protocol):
+    """Protocol for simple I/O reader instances.
+
+    This protocol only supports blocking I/O.
+    """
+
+    __slots__ = ()
+
+    def read(self, size: int = ..., /) -> T:
+        """Read data from the I/O stream and return it.
```

Comment:
I think we shouldn't include those comments here (see the other protocols). In addition, I think we should mark those methods as `@abstractmethod`.

Reply:
What's the harm of including the docstrings? I find it rather helpful to have those available at runtime and in IDEs. I would even prefer the other protocols to have a docstring.

I'm not sure about using `@abstractmethod`. The other protocols are called "ABCs" in the documentation and I think the methods being marked as abstract are holdovers from the "real" ABCs in `collections.abc`, not necessarily a practice we want to continue. See also the protocols in `wsgiref.types`, where the methods are not marked as abstract either.

Reply:
I agree with including docstrings. They are also used in `help()`, for example.

As for abstractmethod, wouldn't it make a difference in cases where a class inherits explicitly from the protocol? If we mark the methods explicitly, we'll stop people from inheriting the protocol without overriding these methods.

Reply:
Right, it slipped my mind that Protocols all have `ABCMeta` as (indirect) meta class, so it makes sense to add it.

(This actually means that basically all protocols should have their methods marked with `@abstractmethod` at runtime. It should also probably be [documented](https://docs.python.org/3/library/typing.html#typing.Protocol) and the examples should also use the decorator. But those are separate issues.)